### ISSUE TITLE
Buttons without link have empty <a> tag, use <div> instead

### ISSUE THREAD

**Issue by sethmclellana8c** [WordPress/gutenberg#65421]
### Description

Buttons without a link set currently have an empty `<a>` tag. But it should default to not having that tag at all and just use a `<div>`.

### Step-by-step reproduction instructions

1.  I spun up a demo site on instawp.io and the default Twenty Twenty-Four theme was active.
2. Went through the first [checks for accessibility](https://make.wordpress.org/themes/handbook/review/accessibility/required/).
3. Tabbed through the homepage to do the check for "Keyboard navigation"
4. Tabbing through the homepage hits the top elements on the page, but skips the "About us" button in top section.
5. Talked with @amberhinds and she suggested this isn't an accessibility fail, but rather an improvement to the block logic

### Environment info

* Device: Mac Laptop
* OS: 14.6.1 (23G93)
* Browser: "Chrome Version 128.0.6613.138 (Official Build) (arm64)"
* WP version: 6.6.2

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by carolinan** [comment]
@stephdau HI, can you please update the description with the explanation _why_ the empty `<a>` tag should be replaced with a `<div>`.

**Comment by carolinan** [comment]
@sethmclellana8c  HI, can you please update the description with the explanation **why** the empty `<a>` tag should be replaced with a `<div>`.

**Comment by amberhinds** [comment]
@carolinan this is related to @joedolson's comment on https://github.com/WordPress/gutenberg/issues/63534.

**Comment by carolinan** [comment]
I am not asking for myself.
Any request for a change will be easier to consider if it has a motivation and explanation. Let's not make the assumption that all contributors understand what the problem is and **why** it needs to change.

**Comment by ellatrix** [comment]
I don't get this. Why leave a button without a link? I don't think we should optimise for leaving blank options. Perhaps add a pre-publish check that warns people about the empty link instead? That seems actually useful instead of silently failing?

**Comment by amberhinds** [comment]
@ellatrix unless we're going to block post save because of invalidly formatted content (which might be a legitimate choice, but there is no precedent for that in WordPress), even with a warning, we still need to change the output of the element in case people choose to ignore the element.

**Comment by joedolson** [comment]
@ellatrix I'm not sure what you're suggesting. If we have the option to prevent publishing unless problems are fixed, I'm certainly in favor of that; but practically speaking I think we should produce the element people believe they are using. If they have not added a link to the element, then why would the button behave as a link? They surely aren't expecting that, and are just looking to implement the button design as a characteristic in their site. 

There's no reason that the 'button' element can't be considered a design characteristic, and not require a link; it just needs to be a different element depending on how it is used.

**Comment by t-hamano** [comment]
Note that a UI for selecting HTML elements (`a`, `button`) has been added to the Advanced panel. See #70139
